recipe/conda_build_config.yaml

```diff
@@ -1,4 +1,4 @@
 mpi:
-  - nompi
-  - mpich    # [not win and not (osx and arm64)]
-  - openmpi  # [not win and not (osx and arm64)]
+  - nompi 
+  - mpich  # [not win]
```

Comment:
Should we pin `mpich` and `openmpi`? It might also be interesting to document why one is needed on windows while the other is needed on the other platforms.

Reply:
The stuff there is used as a matrix so artefacts are built for each entry. We just only use nompi for windows. 
The reason why we do not build with openmpi on anything apart for osx-arm64 is because there seem to be some dependency conflict on all archs except for osx-arm64. The moment I switch from openmpi to mpich - the conflict goes away... A separate ticket will be created to investigate. It seems that the igraph is pretty urgent, and arpack is its dependency. Therefore I thought we could investigate the issue with openmpi later. @sumit0190  would be good to hear your opinion on this too, please.

Reply:
For now we should just document why `openmpi` isn't being used on anything else other than `osx-arm64`. The `windows` part probably doesn't need any clarification since no MPI implementation other than the one offered by MS is available.